### ISSUE TITLE
5.15 Kernel module is not supported

### ISSUE THREAD

**Issue by Xuebit** [DisplayLink/evdi#333]
* Are you using the latest driver?
Yes
  * Are you using the latest EVDI version?
Yes, 1.9.1
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?
Yes
  * Is this issue related to evdi/kernel?
Yes

Distribution = Ubuntu 21.10
Kernel = 5.15.5-76051505
Xorg = 2:1.20.13
Desktop environment = awesomewm

Logs and Description

5.15 kernel does not load the dkms-evdi module:

Building module:
cleaning build area...
make -j16 KERNELRELEASE=5.15.5-76051505-generic -C /lib/modules/5.15.5-76051505-generic/build M=/var/lib/dkms/evdi/1.9.1/build DKMS_BUILD=1...(bad exit sta
tus: 2)
ERROR (dkms apport): kernel package linux-headers-5.15.5-76051505-generic is not supported
Error! Bad return status for module build on kernel: 5.15.5-76051505-generic (x86_64)
Consult /var/lib/dkms/evdi/1.9.1/build/make.log for more information.
   ...done.
/etc/kernel/postinst.d/initramfs-tools:
update-initramfs: Generating /boot/initrd.img-5.15.5-76051505-generic

/var/lib/dkms/evdi/1.9.1/build/make.log:
https://pastebin.com/u65Yn2Sr

**Comment by timnolte** [comment]
I'm pretty sure this is related to this: https://github.com/DisplayLink/evdi/issues/308

**Comment by cdonatom** [comment]
Have you tried to compile the module from devel branch?

**Comment by marcelometal** [comment]
Try this steps, it worked for me:
https://displaylink.org/forum/showpost.php?p=92453&postcount=3

After installing:
https://github.com/AdnanHodzic/displaylink-debian/blob/master/post-install-guide.md#display-detection

**Comment by speedo-spin** [comment]
Also does not work for me on Arch linux. Followed by this [wiki](https://wiki.archlinux.org/title/DisplayLink).
I was able to get it work by downgrading to 5.10.84-1 kernel and installing `evdi`.

However the same system config on 5.15.7-1 Kernel and `evdi-git` instead of `evdi` does not work.
`xrandr --listproviders` gives no modesetting provider
```
Providers: number : 1
Provider 0: id: 0x56 cap: 0xf, Source Output, Sink Output, Source Offload, Sink Offload crtcs: 4 outputs: 4 associated providers: 0 name:Unknown AMD Radeon GPU @ pci:0000:05:00.0
```

`inxi -Gx`
```
Graphics:
  Device-1: AMD Cezanne vendor: Hewlett-Packard driver: amdgpu v: kernel
    bus-ID: 05:00.0
  Device-2: DisplayLink MB16AC type: USB driver: N/A bus-ID: 1-2.2:6
  Device-3: Logitech Webcam C310 type: USB driver: snd-usb-audio,uvcvideo
    bus-ID: 3-2.3.6:6
  Device-4: Luxvisions Innotech Limited HP HD Camera type: USB
    driver: uvcvideo bus-ID: 3-4:3
  Display: x11 server: X.Org 1.21.1.2 driver: loaded: amdgpu,ati
    unloaded: modesetting resolution: 1: 1920x1080~60Hz 2: 1920x1200~60Hz
  OpenGL: renderer: AMD RENOIR (DRM 3.42.0 5.15.7-1-MANJARO LLVM 13.0.0)
    v: 4.6 Mesa 21.2.5 direct render: Yes
```

**Comment by speedo-spin** [comment]
This fork works for me https://github.com/nilathedragon/evdi on `5.15.7-1` kernel

**Comment by zachspar** [comment]
> This fork works for me https://github.com/nilathedragon/evdi on `5.15.7-1` kernel

Me too :) this is the way to go for an interim solution. @DisplayLink-Admin when will we have a fix for this, maybe we can open a PR from this fix to your repo.

**Comment by displaylink-emajewsk** [comment]
This has been fixed in evdi 1.10

**Comment by zachspar** [comment]
> This has been fixed in evdi 1.10

Why isn't it bundled with the DisplayLink driver? I am running fedora and am required to manually sign evdi each time I update.

**Comment by displaylink-emajewsk** [comment]
@zachspar It is bundled with the 5.5 driver: https://www.synaptics.com/products/displaylink-graphics/downloads/ubuntu
